src/irc.py

```diff
@@ -4,106 +4,104 @@
 """IRC Bridge charm business logic."""
 
 import logging
-import os
-import pathlib
 import shutil
-import tempfile
-import time
-import typing
+import subprocess  # nosec
 
+import yaml
+from charms.operator_libs_linux.v1 import systemd
 from charms.operator_libs_linux.v2 import snap
 
-import constants
 import exceptions
+from charm_types import CharmConfig, DatasourceMatrix, DatasourcePostgreSQL
+from constants import (
+    IRC_BRIDGE_CONFIG_DIR_PATH,
+    IRC_BRIDGE_CONFIG_FILE_PATH,
+    IRC_BRIDGE_HEALTH_PORT,
+    IRC_BRIDGE_KEY_ALGO,
+    IRC_BRIDGE_KEY_OPTS,
+    IRC_BRIDGE_PEM_FILE_PATH,
+    IRC_BRIDGE_REGISTRATION_FILE_PATH,
+    IRC_BRIDGE_SNAP_NAME,
+    IRC_BRIDGE_TARGET_FILE_PATH,
+    IRC_BRIDGE_TEMPLATE_CONFIG_FILE_PATH,
+    IRC_BRIDGE_TEMPLATE_TARGET_FILE_PATH,
+    IRC_BRIDGE_TEMPLATE_UNIT_FILE_PATH,
+    IRC_BRIDGE_UNIT_FILE_PATH,
+    SNAP_PACKAGES,
+    SYSTEMD_DIR_PATH,
+)
 
 logger = logging.getLogger(__name__)
 
 
-class ReloadError(exceptions.SnapError):
+class ReloadError(exceptions.SystemdError):
     """Exception raised when unable to reload the service."""
 
 
-class StartError(exceptions.SnapError):
+class StartError(exceptions.SystemdError):
     """Exception raised when unable to start the service."""
 
 
-class StopError(exceptions.SnapError):
+class StopError(exceptions.SystemdError):
     """Exception raised when unable to stop the service."""
 
 
 class InstallError(exceptions.SnapError):
     """Exception raised when unable to install dependencies for the service."""
 
 
-class IRCBRidgeService:
+class IRCBridgeService:
```

Comment:
There are not instance or class attributes nor constructor. I'm wondering if this shouldn't simply be a module ?

Reply:
It could, that's in spirit of the bind-operator, it has a similar class (BindService) without a constructor or attributes. Propose to leave it as is for now.

Reply:
Ok, but let's track the change in a ticket